File: backend/schemas/review.py
from datetime import datetime

from pydantic import BaseModel, Field


class ReviewCreate(BaseModel):
    booking_id: int
    rating: int = Field(ge=1, le=5)
    comment: str = Field(min_length=2, max_length=1000)


class ReviewOut(BaseModel):
    id: int
    customer_id: int
    provider_id: int
    booking_id: int | None = None
    rating: int
    comment: str | None = None
    created_at: datetime
    customer_name: str | None = None
    provider_name: str | None = None

File: frontend/src/types/review.ts
export type Review = {
  id: number;
  customer_id: number;
  provider_id: number;
  booking_id?: number | null;
  rating: number;
  comment?: string | null;
  created_at: string;
  customer_name?: string | null;
  provider_name?: string | null;
};

export type ReviewPayload = {
  booking_id: number;
  rating: number;
  comment: string;
};
